linetools/lists/linelist.py

```diff
@@ -344,7 +349,195 @@ def all_transitions(self,line):
+        cond = (wv_obs >= wvlims[0]) & (wv_obs <= wvlims[1])
+        if np.sum(cond) == 0:
+            if verbose:
+                print('[strongest_transitions] Warning: no transitions found within wvlims; returning None')
+            return None
+        elif isinstance(data,dict): #Only 1 case from a dict format
+            return data
+        elif np.sum(cond) == 1: #only 1 case from a QTable format
+            name = data[cond]['name'][0]
+            return self.__getitem__(name)
+        else:
+            #remove transitions out of range
+            data = data[cond]
+            #add strength column to data
+            strength = data['wrest'] * data['f']
+            data.add_column(Column(data=strength,name='strength'))
+            data.sort('strength')
+            data.reverse()
+            #keep only the first n_max or less
+            data = data[:n_max]
+            #remove column strength for consistency
+            data.remove_column('strength')
+            if len(data) == 1: #return a dictionary
+                name = data['name'][0]
+                return self.__getitem__(name)
+            else:
+                return data
+
+    def available_transitions(self, wvlims, z, n_max=100,n_max_tuple=3, min_strength=4):
+        """For a given wavelength range in Angstroms, wvlims=(wv_min,wv_max),
+        this function retrieves the n_max_tuple strongest transitions per each ion 
+        species in the LineList available at redshift z within such a wavelength 
+        range. The output is sorted by strength of the strongest available transition 
+        per ion species, modualted by abundance.
+
+        Parameters:
+        ----------
+        wvlims : tuple of Quantity
+            Wavelength range, e.g. wvlims=(1100*u.AA, 3200*u.AA)
+        z : float
+            Redshift
+        n_max : int, optional
+            Maximum number of transitions retrieved
+        n_max_tuple : int, optional
+            Maximum number of transitions in a given ion species to 
+            retrieve. e.g., if Lyman series are all available, it will 
+            retrieve only up to Lyman gamma if n_max_tuple=3.
+        min_strength : float, optional
+            Minimum strenght calculated from log10(wrest * fosc * abundance)
+            In thin space HI 1215 has 14.7 [not]
+
+        
+        Returns:
+        ----------
+        dict (if only 1 transition found) or QTable (if > 1 transitions are found)
+        or None (if no transition is found)
+        """
+        if all(isinstance(n,int) for n in [n_max,n_max_tuple]):
+            pass
+        else:
+            raise SyntaxError('Both n_max and n_max_tuple must be integers!')
+
+        if n_max == 0:
+            return None
+
+        data = copy.deepcopy(self._data)
+
+        #identify unique ion_names (e.g. HI, CIV, CIII)
+        unique_ion_names = list(set([name.split(' ')[0] for name in data['name']]))
+        unique_ion_names = np.array(unique_ion_names)
```

Comment:
By 'unique', do you mean non-duplicated?  Seems like this array will have duplicates.  You might use np.unique